### ISSUE TITLE
"\0" is invalid JSON

### ISSUE THREAD

**Issue by czechdude** [mgholam/fastJSON#52]
Hi,
we have the same problem as stated here for the Java version
https://github.com/alibaba/fastjson/issues/118

so the rest is just copied:

According the JSON specification, ```"\0"``` is not a valid JSON string and must be encoded with four hexadecimal bytes. It's not just a theoretical problem - some python JSON parsers do not accept it.

I recommend the following unit test:
```
public class FastJsonTest {
    @Test
    public void testEncodeNulByte() throws Exception {
        String json = JSON.toJSONString("\0");
        assertEquals("\"\\u0000\"", json);
    }
}
```

**Comment by mgholam** [comment]
The following works as expected in the tests i.e. you get \u0000
```
    [Test]
    public static void zerostring()
    {
        var s = fastJSON.JSON.ToJSON("\0");
        Console.WriteLine(s);
        var o = fastJSON.JSON.ToObject(s);
        Console.WriteLine("" + o);
    }
```

**Comment by czechdude** [comment]
but wait that is wrong!

I wrote a test and it does not do what is expected:

```
	    [Test]
	    public static void SerializeStringWithZeroEscapeTest()
	    {

	        fastJSON.JSONParameters param = new JSONParameters();
	        param.SerializeNullValues = false;
	        param.UseExtensions = false;
	        param.UseFastGuid = false;
	        param.UseEscapedUnicode = true;
	        var testClass01 = new TestClass01();
	        testClass01.stringid = "Test\0Test";

	        var newJson = JSON.ToJSON(testClass01,param);
	        Assert.True(newJson.Contains("\\u0000"), "wrong repre of \\0");

	        param = new JSONParameters();
	        param.SerializeNullValues = false;
	        param.UseExtensions = false;
	        param.UseFastGuid = false;
	        param.UseEscapedUnicode = false;
	        testClass01 = new TestClass01();
	        testClass01.stringid = "L6645230045235\00175";

	        newJson = JSON.ToJSON(testClass01, param);
	        Assert.True(newJson.Contains("L6645230045235\\00175"), "wrong repre of without escaping unicode \\0");
        }
```

**Comment by mgholam** [comment]
Fixed in v2.1.25
